### DISCUSSION TITLE
micronaut-platform 4.4.1 jackson dependencies are not matching

### DISCUSSION THREAD

**Discussion by ethanhere** [micronaut-projects/micronaut-core#10790] [Q&A] [answered]
We use micronaut current latest release 4.4.1 (https://github.com/micronaut-projects/micronaut-platform/releases as on 05/02/2024)
This release has micronaut-core 4.4.6. It's libs.versions.toml has:
managed-jackson = "2.16.1"
managed-jackson-databind = "2.17.0"

Our application pom has dependency management micronaut-platform 4.4.1
    <dependencyManagement>
        <dependencies>

            <dependency>
                <groupId>io.micronaut.platform</groupId>
                <artifactId>micronaut-platform</artifactId>
                <version>4.4.1</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>

            <dependency>
                <groupId>io.micronaut</groupId>
                <artifactId>micronaut-core-bom</artifactId>
                <version>4.4.6</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>

And our Jackson dependencies follows micronaut 4.4.1's Jackson versions:
Jackson-databind: 2.17.0
Rest: 2.16.1

In our app, when we call 
JsonParser parser;

final JsonNode root = parser.readValueAsTree();

We got error:
java.lang.NoSuchMethodError: 'com.fasterxml.jackson.core.JsonParser$NumberTypeFP com.fasterxml.jackson.core.JsonParser.getNumberTypeFP()'
	at com.fasterxml.jackson.databind.deser.std.BaseNodeDeserializer._fromFloat(JsonNodeDeserializer.java:765)
	at com.fasterxml.jackson.databind.deser.std.BaseNodeDeserializer._deserializeContainerNoRecursion(JsonNodeDeserializer.java:578)
	at com.fasterxml.jackson.databind.deser.std.JsonNodeDeserializer.deserialize(JsonNodeDeserializer.java:100)
	at com.fasterxml.jackson.databind.deser.std.JsonNodeDeserializer.deserialize(JsonNodeDeserializer.java:25)
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:342)
	at com.fasterxml.jackson.databind.ObjectMapper._readValue(ObjectMapper.java:4881)
	at com.fasterxml.jackson.databind.ObjectMapper.readTree(ObjectMapper.java:3145)
	at com.fasterxml.jackson.core.JsonParser.readValueAsTree(JsonParser.java:2455)

BaseNodeDeserializer lives in Jackson-databind 2.17.0, in that jar, BaseNodeDeserializer._fromFloat method will call:
    protected final JsonNode _fromFloat(JsonParser p, DeserializationContext ctxt, JsonNodeFactory nodeFactory) throws IOException {
			
		JsonParser.NumberTypeFP nt = p.getNumberTypeFP(); 

JsonParser lives in Jackson-core 2.16.1 and it doesn't have such method.

Should micronaut use consistent Jackson dependencies? How should we resolve the issue?

**Comment by yawkat** [comment, marked as answer]
Looks like #10745

**Reply by ethanhere** [reply]
I commented on https://github.com/micronaut-projects/micronaut-core/issues/10745. When or which release can we expect a fix of this? I think the potential risk is substancial because micronaut lives in application jvm and user's business logic could call aross jackson jars.

In addition, I hope micronaut exam dependencies it uses, and make sure they are compatible.

**Reply by sdelamo** [reply]
> I commented on https://github.com/micronaut-projects/micronaut-core/issues/10745. When or which release can we expect a fix of this? I think the potential risk is substancial because micronaut lives in application jvm and user's business logic could call aross jackson jars.


We created a PR: 

https://github.com/micronaut-projects/micronaut-core/pull/10795 

and a release will happen shortly. 


> In addition, I hope micronaut exam dependencies it uses, and make sure they are compatible.

Yes, we exam and  test dependencies. However, we missed this case.
